### PR TITLE
Add user registration to cli

### PR DESCRIPTION
**Pull request by robigan**
This PR adds a custom function to be able to add users from the CLI. This is a bare-bones modification to cli.py that works, and I've semi tested (against a production environment lol). I am opening this PR as an RFC.

### CODE REVIEW

File: backend/cli.py [hunk 1]
```diff
@@ -57,8 +65,27 @@ def hash_password_cmd() -> None:
...
+        password_hash=hash_password(password),
+        name=name,
+        created=datetime_utc.now(),
+        account_type=UserAccountType.REGULAR,
+    )
+    if await check_whether_email_is_in_use(email):
+        raise AssertionError("Email address already in use")
+    user_created = await create_user(user)
+    assert_some(user_created.id)
+    print("Created user with id:", user_created.id)
```
Comment (evroon) [inline backend/cli.py:85]:
```suggestion
    logger.info(f"Created user with id: {user_created.id}")
```

File: backend/cli.py [hunk 2]
```diff
@@ -57,8 +65,27 @@ def hash_password_cmd() -> None:
...
+async def register_user(email, password, name) -> None:
+    user = User(
+        email=email,
+        password_hash=hash_password(password),
+        name=name,
+        created=datetime_utc.now(),
+        account_type=UserAccountType.REGULAR,
+    )
+    if await check_whether_email_is_in_use(email):
+        raise AssertionError("Email address already in use")
```
Comment (evroon) [inline backend/cli.py:82]:
```suggestion
        logger.error("Email address already in use")
        raise SystemExit(1)
```

File: backend/cli.py [hunk 3]
```diff
@@ -57,8 +65,27 @@ def hash_password_cmd() -> None:
 async def create_dev_db() -> None:
     await sql_create_dev_db()
 
+@click.command()
+@click.option('--email', prompt='Email', help='The email used to log into the account.')
+@click.option('--password', prompt='Password', help='The password used to log into the account.')
+@click.option('--name', prompt='Name', help='The name associated with the account.')
+@run_async
+async def register_user(email, password, name) -> None:
```
Comment (evroon) [inline backend/cli.py:73]: There are types missing here: https://github.com/robigan/bracket/actions/runs/7989651896/job/21816653345
```suggestion
async def register_user(email: str, password: str, name: str) -> None:
```

**Review by evroon — COMMENTED** [review]
Thanks, this is useful!
A few suggestions.
Could you also make sure the CI succeeds? https://github.com/robigan/bracket/actions/runs/7989651896/job/21816653345

**Comment by robigan** [comment]
> Thanks, this is useful! A few suggestions. Could you also make sure the CI succeeds? https://github.com/robigan/bracket/actions/runs/7989651896/job/21816653345

I just actually made the modifications based on pre-existing code, applied the changes to a production machine (as I needed a separate account) and they had seemed to have worked. I'll install the dev tool chain and I'll run the checks.

**Review by evroon — APPROVED** [review]
LGTM!

**Pull request merged**